<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MarginCalc</class>
 <widget class="QDialog" name="MarginCalc">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <family>Arial</family>
    <pointsize>10</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_15">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_39">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_13">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_40">
         <item>
          <widget class="QLabel" name="label_29">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Netlist file :                                </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditNetlist_3">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="whatsThis">
            <string>Netlist file with the handle for parameters.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonNetlist_3">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_41">
         <item>
          <widget class="QLabel" name="label_30">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Reference file for netlist outputs :</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditSolution_3">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="whatsThis">
            <string>at least one solution point for the reference of the analyzer. More solutions give accurate result.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonSolution_3">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_42">
         <item>
          <widget class="QLabel" name="label_31">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>File of netlist variables  :             </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditParam_3">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="whatsThis">
            <string>The parameters with minimum, stepsize and maximum value.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonParam_3">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_43">
         <item>
          <widget class="QLabel" name="label_32">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Temporary folder :                     </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditTempF_3">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="whatsThis">
            <string>Temporary folder for storing simulation files. The folder size could get large!</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonTempF_3">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_44">
         <item>
          <widget class="QLabel" name="label_33">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Output folder :                           </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditOutF_3">
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="whatsThis">
            <string>Output files are stored here.</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="toolButtonOutF_3">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_14">
       <item>
        <widget class="QDialogButtonBox" name="buttonBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="standardButtons">
          <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::Reset</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxNoise_4">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="whatsThis">
          <string>If checked, Resistors will have thermal noise sources on them.</string>
         </property>
         <property name="text">
          <string>Magnetic noise</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBoxNoise_3">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="font">
          <font>
           <family>Arial</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="whatsThis">
          <string>If checked, Resistors will have thermal noise sources on them.</string>
         </property>
         <property name="text">
          <string>Thermal noise</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_38">
     <item>
      <widget class="QLabel" name="label_Picture_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>900</width>
         <height>400</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="pixmap">
        <pixmap>:/Help/Help/Margin.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_Text_3">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>11</pointsize>
         <weight>50</weight>
         <bold>false</bold>
        </font>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>Margin calculation method:
Nominal margins are set to [-90% - +200%]
One parameter is swept while all other 
parameters remain at nominal value. 
The margin analyzer stops when the 
step to the next parameter value is below 
the specified maximum margin uncertainty. </string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="TestButton_3">
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Simulate netlist 
with nominal values</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="whatsThis">
      <string>Parameters for analyzer</string>
     </property>
     <property name="title">
      <string>Settings</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_31">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_11">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_32">
          <item>
           <widget class="QLabel" name="label_25">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Maximum uncertainty of margins [%] : </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditErrorPer_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="whatsThis">
             <string>The error percentage acceptable for analyzis of the grayzone. This value effect the speed of simulation!</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_33">
          <item>
           <widget class="QLabel" name="label_27">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Temperature [K]:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditTempVal_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="whatsThis">
             <string>Ambient temperature of the chip surface.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_34">
          <item>
           <widget class="QLabel" name="label_28">
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Rn*Ic (uV) :</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditRnIc_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="whatsThis">
             <string>This value depends on your fabrication process. Leave the default value if you are not sure.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_12">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_35">
          <item>
           <widget class="QLabel" name="label_26">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Number of iterations :</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditIterNum_3">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="whatsThis">
             <string>Number of iterations for optimizer program. this number effect the speed directly.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_36">
          <item>
           <widget class="QLabel" name="label_24">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Standard deviation of junctions [%]:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditJJSpread_3">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="whatsThis">
             <string>Variations of the critical current of the junction base on the foundry data.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_37">
          <item>
           <widget class="QLabel" name="label_23">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Standard deviation of parameter [%] :</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="lineEditParamVar_3">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Arial</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="whatsThis">
             <string>Variations of the parameters based on the data from foundry.</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>MarginCalc</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>MarginCalc</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
